openapi3gen/openapi3gen.go

```diff
@@ -195,14 +195,16 @@ func (g *Generator) generateWithoutSaving(parents []*theTypeInfo, t reflect.Type
 			return nil, &CycleError{}
 		}
 	}
-
-	if cap(parents) == 0 {
+	isRoot := cap(parents) == 0
+	if isRoot {
 		parents = make([]*theTypeInfo, 0, 4)
 	}
 	parents = append(parents, typeInfo)
 
+	isNullable := false
 	for t.Kind() == reflect.Ptr {
 		t = t.Elem()
+		isNullable = !isRoot
```

Comment:
I assume we don't want to mark root structs `nullable` so I we have this extra check here.